app/page.tsx

```diff
@@ -58,7 +58,25 @@ function Cookies() {
     const parsedCookies = parseCookies(); // get all stored cookies
     
     Object.values(parsedCookies).forEach(cookieData => {
-      initialCookies.push(JSON.parse(cookieData));
+      try {
+        const cookie = JSON.parse(cookieData)
+
+        // Check cookie is a valid Cookie
+        if (
+          typeof cookie.id !== 'number' ||
+          typeof cookie.imgSrc !== 'string' ||
+          typeof cookie.centerX !== 'number' ||
+          typeof cookie.centerY !== 'number'
+        ) {
+          throw new Error('Invalid Cookie');
+        } else {
```

Comment:
I don't think `else` is needed here since it throws.

Reply:
Ah yes, that isn't needed.